dspace-api/src/main/java/org/dspace/content/MetadataFieldServiceImpl.java

```diff
@@ -94,6 +98,20 @@ public MetadataField findByElement(Context context, String metadataSchemaName, S
         return metadataFieldDAO.findByElement(context, metadataSchemaName, element, qualifier);
     }
 
+    @Override
+    public MetadataField findByString(Context context, String mdString, char separator) throws SQLException {
+        String[] seq = StringUtils.split(mdString, separator);
+        String schema = seq.length > 1 ? seq[0] : null;
+        String element = seq.length > 1 ? seq[1] : null;
+        String qualifier = seq.length == 3 ? seq[2] : null;
+        if (schema == null || element == null) {
+            return null;
+        } else {
+            MetadataSchema metadataSchema = metadataSchemaService.find(context, schema);
+            return this.findByElement(context, metadataSchema, element, qualifier);
```

Comment:
There's no need to get the `MetadataSchema` object first, you can just use the other `findByElement()` method which takes in the `metadataSchemaName`: https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/content/MetadataFieldServiceImpl.java#L92